superbench/common/utils/process.py

```diff
@@ -4,19 +4,47 @@
 """Process Utility."""
 
 import subprocess
+import os
+import shlex
 
+from superbench.common.utils import stdout_logger
 
-def run_command(command):
+
+def run_command(command, quite=False, flush_output=False):
     """Run command in string format, return the result with stdout and stderr.
 
     Args:
         command (str): command to run.
+        quite (bool): no stdout display of the command if quite is True.
+        flush_output (bool): enable real-time output flush or not when running the command.
 
     Return:
         result (subprocess.CompletedProcess): The return value from subprocess.run().
     """
-    result = subprocess.run(
-        command, stdout=subprocess.PIPE, stderr=subprocess.STDOUT, shell=True, check=False, universal_newlines=True
-    )
-
-    return result
+    if flush_output:
+        process = None
+        try:
+            args = shlex.split(command)
+            process = subprocess.Popen(
+                args, cwd=os.getcwd(), stdout=subprocess.PIPE, stderr=subprocess.STDOUT, universal_newlines=True
+            )
+            output = ''
+            for line in process.stdout:
+                output += line
+                if not quite:
+                    stdout_logger.log(line)
+            process.wait()
+            retcode = process.poll()
+            return subprocess.CompletedProcess(args=args, returncode=retcode, stdout=output, stderr=output)
```

Comment:
no need for stderr=output? it's duplicate

Reply:
removed